### PR TITLE
daml-lf: introduce Numeric in internal AST

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This PR is part of the effort to implement variable scaled Numeric type (See #2289)

In This PR we introduce Numeric in the internal  AST together with the nat kind and types.
We update the decoder/encoder to read write DAML-LF proto and the type checker.

In practice, the engine still assumes all the Numeric are of scale 10 (the old fixed decimal scale).

This PR does not introduce any externally visible change. 

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [X] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by remyhaemmerle-da** [comment]
> We should not merge this PR until the damlc integration test for DAML-LF 1.dev works again. I'm happy to help with that.

I reenable the dev intgegration test.

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala [hunk 1]
```diff
@@ -280,12 +280,19 @@ private[archive] class DecodeV1(minor: LV.Minor) extends Decode.OfPackage[PLF.Pa
...
           val prim = lfType.getPrim
-          val (tPrim, minVersion) = primTypeTable(prim.getPrim)
-          assertSince(minVersion, prim.getPrim.getValueDescriptor.getFullName)
-          (TBuiltin(tPrim) /: [Type] prim.getArgsList.asScala)((typ, arg) =>
-            TApp(typ, decodeType(arg)))
+          if (prim.getPrim == PLF.PrimType.DECIMAL) {
+            // FixMe: https://github.com/digital-asset/daml/issues/2289
+            //  enable the check once the compiler produces proper DAML-LF 1.dev
+            // assertUntil(LV.Features.numeric, "PLF.PrimType.DECIMAL")
+            TDecimal
```
Comment (gerolf-da) [inline daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala]: apply to the argument.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/package.scala [hunk 2]
```diff
@@ -25,7 +25,10 @@ package object tests {
     "DivulgenceIT" -> (new Divulgence(_)),
     "IdentityIT" -> (new Identity(_)),
     "TimeIT" -> (new Time(_)),
-    "ContractKeysSubmitterIsMaintainerIT" -> (new ContractKeysSubmitterIsMaintainer(_)),
+    // FixMe: https://github.com/digital-asset/daml/issues/2289
+    //   enable once the testing dar is compiled using 1.dev
+    //   (see ledger/test-common/BUILD.bazel)
+    //   "ContractKeysSubmitterIsMaintainerIT" -> (new ContractKeysSubmitterIsMaintainer(_)),
```
Comment (gerolf-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/package.scala]: Uncomment this, as it should work now.

File: ledger/test-common/BUILD.bazel [hunk 3]
```diff
@@ -60,7 +60,10 @@ da_scala_library(
 
 lf_test_versions = [
     (lf_stable_version, "stable"),
-    ("1.dev", "dev"),
+    # FixMe: https://github.com/digital-asset/daml/issues/2289
+    #  switch back to 1.dev once numerics is done.
+    #  1.dev is currently broken, then we compile against 1.6
+    ("1.6", "dev"),
```
Comment (gerolf-da) [inline ledger/test-common/BUILD.bazel]: Set it to `1.dev`.

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala]: done

File: ledger/test-common/BUILD.bazel (hunk 3 above)
Comment (remyhaemmerle-da) [reply ledger/test-common/BUILD.bazel]: done

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/package.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/package.scala]: done

**Pull request merged**